src/transformers/models/pix2struct/modeling_pix2struct.py

```diff
@@ -0,0 +1,1686 @@
+        self.dropout = config.attention_dropout
+        self.inner_dim = self.n_heads * self.key_value_proj_dim
+
+        # Mesh TensorFlow initialization to avoid scaling before softmax
+        self.query = nn.Linear(self.hidden_size, self.inner_dim, bias=False)
+        self.key = nn.Linear(self.hidden_size, self.inner_dim, bias=False)
+        self.value = nn.Linear(self.hidden_size, self.inner_dim, bias=False)
+        self.o = nn.Linear(self.inner_dim, self.hidden_size, bias=False)
+
+        self.gradient_checkpointing = False
+
+    def forward(
+        self,
+        hidden_states,
+        attention_mask=None,
+        position_bias=None,
+        layer_head_mask=None,
+        output_attentions=False,
+    ):
+        """
+        Self-attention block
+        """
+        # Input is (batch_size, seq_length, dim)
+        # Mask is (batch_size, key_length) (non-causal) or (batch_size, key_length, key_length)
+        # past_key_value[0] is (batch_size, n_heads, q_len - 1, dim_per_head)
+        batch_size, seq_length = hidden_states.shape[:2]
+
+        def shape(states):
+            """projection"""
+            return states.view(batch_size, -1, self.n_heads, self.key_value_proj_dim).transpose(1, 2)
+
+        def unshape(states):
+            """reshape"""
+            return states.transpose(1, 2).contiguous().view(batch_size, -1, self.inner_dim)
+
+        # get query states
+        query_states = shape(self.query(hidden_states))  # (batch_size, n_heads, seq_length, dim_per_head)
+
+        # get key/value states
+        key_states = shape(self.key(hidden_states))
+        value_states = shape(self.value(hidden_states))
+
+        # compute scores
+        scores = torch.matmul(
+            query_states, key_states.transpose(3, 2)
+        )  # equivalent of torch.einsum("bnqd,bnkd->bnqk", query_states, key_states), compatible with onnx op>9
+
+        if position_bias is None:
+            position_bias = torch.zeros(
+                (1, self.n_heads, seq_length, seq_length), device=scores.device, dtype=scores.dtype
+            )
+            if self.gradient_checkpointing and self.training:
+                position_bias.requires_grad = True
+
+            if attention_mask is not None:
+                if attention_mask.dim() == 2:
+                    position_bias = position_bias + attention_mask[:, None, :, None].to(position_bias.device)
+                else:
+                    position_bias = position_bias + attention_mask.to(
+                        position_bias.device
+                    )  # (batch_size, n_heads, seq_length, key_length)
+
+        position_bias_masked = position_bias
+        if position_bias_masked is not None:
+            # replace 0 by -inf
+            position_bias_masked = position_bias_masked.masked_fill(position_bias_masked == 0, -10000.0)
+            position_bias_masked = position_bias_masked.masked_fill(position_bias_masked != -10000, 0)
```

Comment:
-1000 hard coded can it not be the case?

Reply:
Fixed!